Build failure with ffmpeg 6.0

## Environment
* OS Version: macOS
* Source or binary build? source build as of a363f8edb2b02efcc4ae0e65ea8c154d8b78bfc8 for `ign-common3` branch, and I suspect it applies to other branches as well
  <!-- If binary, which version? -->
  <!-- If source, which branch and what is the latest commit? -->

## Description
* Expected behavior: gz-common should build successfully against ffmpeg 6.0
* Actual behavior: since ffmpeg 6.0 was merged in https://github.com/Homebrew/homebrew-core/pull/124363, the `ign-common3` branch is failing to build with a complaint about "undeclared identifier `AV_CODEC_FLAG_TRUNCATED`"
  * [![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=ignition_common-ci-ign-common3-homebrew-amd64&build=161)](https://build.osrfoundation.org/view/ign-citadel/job/ignition_common-ci-ign-common3-homebrew-amd64/161/) https://build.osrfoundation.org/view/ign-citadel/job/ignition_common-ci-ign-common3-homebrew-amd64/161/

## Steps to reproduce
<!-- Provide steps so we can try to reproduce this issue -->

1. Upgrade your homebrew-core packages on macOS (`brew upgrade`)
2. Build `ign-common3` from source (or run https://build.osrfoundation.org/view/ign-citadel/job/ignition_common-ci-ign-common3-homebrew-amd64/ )

## Output

~~~
av/src/AudioDecoder.cc:348:41: error: use of undeclared identifier 'AV_CODEC_CAP_TRUNCATED'; did you mean 'AV_CODEC_ID_TRUEHD'?
  if (this->data->codec->capabilities & AV_CODEC_CAP_TRUNCATED)
                                        ^~~~~~~~~~~~~~~~~~~~~~
                                        AV_CODEC_ID_TRUEHD
/usr/local/Cellar/ffmpeg/6.0/include/libavcodec/codec_id.h:482:5: note: 'AV_CODEC_ID_TRUEHD' declared here
    AV_CODEC_ID_TRUEHD,
    ^
/Users/jenkins/jenkins-agent/workspace/ignition_common-ci-ign-common3-homebrew-amd64/ign-common/av/src/AudioDecoder.cc:349:36: error: use of undeclared identifier 'AV_CODEC_FLAG_TRUNCATED'
    this->data->codecCtx->flags |= AV_CODEC_FLAG_TRUNCATED;
                                   ^
2 errors generated.
~~~

--- a/av/src/AudioDecoder.cc
+++ b/av/src/AudioDecoder.cc
@@ -345,8 +345,10 @@ bool AudioDecoder::SetFile(const std::string &_filename)
   }
 
 #if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(56, 60, 100)
+#if LIBAVCODEC_VERSION_MAJOR < 60
   if (this->data->codec->capabilities & AV_CODEC_CAP_TRUNCATED)
     this->data->codecCtx->flags |= AV_CODEC_FLAG_TRUNCATED;
+#endif
 #else
   if (this->data->codec->capabilities & CODEC_CAP_TRUNCATED)
     this->data->codecCtx->flags |= CODEC_FLAG_TRUNCATED;

--- a/av/src/Video.cc
+++ b/av/src/Video.cc
@@ -182,8 +182,10 @@ bool Video::Load(const std::string &_filename)
   // Inform the codec that we can handle truncated bitstreams -- i.e.,
   // bitstreams where frame boundaries can fall in the middle of packets
 #if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(56, 60, 100)
+#if LIBAVCODEC_VERSION_MAJOR < 60
   if (codec->capabilities & AV_CODEC_CAP_TRUNCATED)
     this->dataPtr->codecCtx->flags |= AV_CODEC_FLAG_TRUNCATED;
+#endif
 #else
   if (codec->capabilities & CODEC_CAP_TRUNCATED)
     this->dataPtr->codecCtx->flags |= CODEC_FLAG_TRUNCATED;

